### DISCUSSION TITLE
Cloudlog and WSJT-X Improved

### DISCUSSION THREAD

**Discussion by soakland5** [magicbug/Cloudlog#1921] [Q&A]
Does anyone know how to set the api for Cloudlog for WSJT-X integration ? It’s included in the latest improved version. I couldn’t get gridttracker to connect to cloudlog either. Followed all the instructions. Any ideas on either ? Want to log wsjtx qsos to Cloudlog 

thanks!

Scott

**Comment by phl0** [comment]
I am using WSTJ-X improved with Cloudlog and it works. What exactly is your issue?

**Comment by soakland5** [comment]
I would like to know the exact URL necessary to get it going. It keeps
throwing an invalid API key error even though it is copied and pasted from
Cloudlog

Scott

**Reply by phl0** [reply]
> I would like to know the exact URL necessary to get it going. It keeps throwing an invalid API key error even though it is copied and pasted from Cloudlog Scott

It is just the basic URL without the /index.php/api/radio stuff. Is your log HTTPS based? Are you using WSJT-X improved on Windows or Linux?

**Comment by soakland5** [comment]
This is Windows 11. The log is web based - no local database is used. And
what is the basic URL? I’ve been told 127.0.0.1 and / or W1AL.Cloudlog.co.uk
.

**Comment by soakland5** [comment]
Yes it’s wsjt-x improved

**Comment by magicbug** [comment]
URL will be `https://w1al.cloudlog.co.uk`

**Comment by phl0** [comment]
Ok so that will fail because most likely the WSJT-X improved installer did not install the two DLLs needed for SSL access to Cloudlog. See my Tweet here: https://twitter.com/flo_0_/status/1593373753756602371?s=20&t=v5DttgXG_GCt6g5MDdwB1Q.

**Comment by soakland5** [comment]
Thanks for that ! Which SSL addon do I need to install for windows 11? Or
is it built in? If so where are the files ?

**Comment by phl0** [comment]
The thing is it is not built-in. I need to dig out my stuff from the time I made it. QRX.

**Comment by phl0** [comment]
Found it. You can install OpenSSL for Win from here:

https://slproweb.com/products/Win32OpenSSL.html

Be sure to pick the correct architecture. I used Win64 OpenSSL v1.1.1s Light and installed to Windows directory. Should work for you also.

**Comment by soakland5** [comment]
Thanks! Tried it but doesn't work on my Win  11. Using WSJT Improved Plus.
Tried both 32 and 64 bit versions of Open SSL.

COuld it be Firewall or NordVPN related?

**Comment by phl0** [comment]
I don't think so. I would try copying the two DLLs from Windows system directory to wsjtx/bin directory.

**Comment by soakland5** [comment]
Tried that many times, no go. Now the settings in the advanced tab of
WSJT-X have disappeared (the CloudLog ones)!!

I had tried another version and then when I put back the correct WSJT-X
Improved version I still can't get the CloudLog settings to appear.

**Comment by phl0** [comment]
That is sad. Apparently there are various versions of WSJT-X improved and only some of them have Cloudlog support. I do mix them up every time also. As a last chance I would offer to support via remote desktop like TeamViewer or similar. Drop me a message if you agree.

**Comment by soakland5** [comment]
We could do that thanks. I uninstalled wsjtx improved and reinstalled it
and got the Cloudlog section back again. But still the API key isn’t
recognized. Would turning off firewalls and VPN help ?

**Comment by phl0** [comment]
No I don't think so. I bet that it is the issue with SSL as it was also discovered by others. As it is normal HTTPS traffic and outgoing I think this is not affected by Firewall/VPN.

**Comment by soakland5** [comment]
Perhaps there is an alternate way to do this ? I use N3FJP’s AC Log as my
primary but really like Cloudlog. Save qso from WSJT-x to Cloudlog……

**Comment by phl0** [comment]
I am sure that there are other ways. But I can only speak about the combination of WSJT-X and Cloudlog. I am sorry. Maybe others can comment here.

**Comment by soakland5** [comment]
I tried using the Cloudlog API with Gridtracker and WSJT-X and neither
worked.

**Comment by AndreasK79** [comment]
I have both working here.

Try this url in both: https://w1al.cloudlog.co.uk/index.php/api/qso.

Test didn't work with that url in WSJT-X improved for me. Removing /api/qso did, but no qso was logged.
Gridtracker worked on testing and logging with api/qso in the url.

**Comment by soakland5** [comment]
I tried that in Gridtracker and it worked !! Thank you !.

**Comment by phl0** [comment]
I am investigating this issue. Seems the Windows buold needs some extra brush up here ... @AndreasK79 assisted with finding a bug. I am trying to find a solution now.

**Comment by phl0** [comment]
Hej @soakland5 with support of @AndreasK79 and @hugosilvaparagramanet I was able to track down the issue and submitted two fixes for WSJT-X improved to cope with issues on entering the Cloudlog URL. Trailing slashes and paths were causing the hickups. I added some code that clean the URL up. Also added a tooltip for the URL field. First patch is included in the builds already. Second one hopefully soon. Feel free to give it a second chance once the binaries / installers are built.

**Comment by soakland5** [comment]
Awesome, thanks for the update!

I’m Using Gridtracker for now to upload as that works fine now thanks to
the tip from the other day
